SEARCH-2076 aggregated search docs improvements

howard opened the pull request.

howard committed:
Refer to both JS and API approaches for search analytics

howard committed:
Fix some broken links in search articles

howard committed:
Document default search page size limit

howard committed:
Add search specifics to variation documentation

howard committed:
Try to fix includes

howard committed:
Fix typo

howard committed:
Add search specifics to multi currency guide

jwittorf commented on nosto-default-service-limits/nosto-service-limits.md:
```diff
@@ -2,18 +2,19 @@
 | Number of Facets       | Up to 100                                                       |
 | Variations             | 100 variation ID's per account                                  |
 | Pins                   | 1.000 per rule                                                  |
+| Products per page      | Up to 250 per request by default                                |
```
```suggestion
| Products per page      | Up to 250 per request by default (please use [../implementing-nosto/implement-search/implement-search-using-api/implementing-search-page#pagination-and-size](pagination))                               |
```

howard committed:
Implement first wave of feedback

Thanks Julian!

howard committed:
Fix some broken includes

howard committed:
Improve pagination/sort/filter docs

Thanks Dom!

howard committed:
Consistently refer to merchandising rules as such

Just "rules" can be confusing to someone whose brain isn't thoroughly
nostified.

tomazvila reviewed (approved):
👍
